src/v/redpanda/admin/security.cc

```diff
@@ -206,6 +206,15 @@ security::role_name parse_role_definition(const json::Document& doc) {
     return role_name;
 }
 
+inline std::unique_ptr<ss::http::reply> make_json_response(
+  std::unique_ptr<ss::http::reply> rep,
+  ss::http::reply::status_type status,
+  ss::json::json_return_type body) {
+    rep->set_status(status);
+    rep->write_body("json", body._res);
+    return rep;
+}
```

Comment:
optional: you can avoid the `json_return_type` to make things a bit less wordy at the call site:

```suggestion
template<class T>
requires std::is_base_of_v<ss::json::jsonable, std::remove_cvref_t<T>>
inline std::unique_ptr<ss::http::reply> make_json_response(
  std::unique_ptr<ss::http::reply> rep,
  ss::http::reply::status_type status,
  T&& body) {
    rep->set_status(status);
    rep->write_body(
      "json", ss::json::formatter::to_json(std::forward<T>(body)));
    return rep;
}
```

Reply:
Nice!